sdl-core/src/main/scala/io/smartdatalake/workflow/action/spark/customlogic/CustomDsTransformer.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2022 ELCA Informatique SA (<https://www.elca.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.smartdatalake.workflow.action.spark.customlogic
+
+
+import io.smartdatalake.util.hdfs.PartitionValues
+import org.apache.spark.sql.{DataFrame, Dataset, Encoder, SparkSession}
+
+import scala.reflect.runtime.universe.TypeTag
+
+/**
+ * Interface to define a custom Spark-Dataset transformation (1:1)
+ * When you implement this interface, you need to provide two case classes: One for your input Dataset
+ * and one for your output Dataset.
+ */
+trait CustomDsTransformer[In <: scala.Product, Out <: scala.Product] extends Serializable {
+  implicit def outputEncoder(implicit typeTag: TypeTag[Out]): Encoder[Out] = org.apache.spark.sql.Encoders.product[Out]
```

Comment:
The encoder instance can be retrieved like that. But this is already implicitly available - so this function is actionally superfluous... where is it used?

Reply:
It will be used by the user's code. It allows the user to map his data back to a DataSet of the Output Datatype.
Concretely, I need this so that my Unit Tests compiles in the File ScalaClassSparkDsTransformerTest at line 44:
inputDS.withColumn("doubled_rating", col("rating") * 2).**as[OutputDataSet]**